src/events/requests/NewRequestEventHandler.ts

```diff
@@ -1,22 +1,77 @@
-import { Message } from 'discord.js';
+import { Message, TextChannel, RichEmbed } from 'discord.js';
 import * as log4js from 'log4js';
 import EventHandler from '../EventHandler';
+import BotConfig from '../../BotConfig';
+import { ReactionsUtil } from '../../util/ReactionsUtil';
+import MentionCommand from '../../commands/MentionCommand';
 
 export default class NewRequestEventHandler implements EventHandler {
 	public readonly eventName = '';
 
 	private logger = log4js.getLogger( 'NewRequestEventHandler' );
 
+	/**
+	 * A map from request channel IDs to internal channel objects.
+	 */
+	private readonly internalChannels: Map<string, TextChannel>;
+
+	constructor( internalChannels: Map<string, TextChannel> ) {
+		this.internalChannels = internalChannels;
+	}
+
 	// This syntax is used to ensure that `this` refers to the `NewRequestEventHandler` object
-	public onEvent = ( message: Message ): void => {
-		this.logger.info( `User ${ message.author.tag } posted a new request to requests channel ${ message.channel.id }` );
+	public onEvent = async ( origin: Message ): Promise<void> => {
+		if ( origin.type !== 'DEFAULT' ) {
```

Comment:
Is it still necessary to delete system messages here? After all, the bot no longer pins messages, does it?

Reply:
Though it could react the waiting emoji to system messages. Might change the behavior to simply ignoring them instead of deleting them?

Reply:
Yeah, I think the system messages should just be ignored altogether, after all we don't do anything with them anymore, do we?